llvm/docs/Contributing.rst

```diff
@@ -124,12 +124,24 @@ For developers to commit changes from Git
 -----------------------------------------
 
 Once a patch is reviewed, you can select the "Squash and merge" button in the
-GitHub web interface. You might need to rebase your change before pushing
-it to the repo.
+GitHub web interface.
 
-LLVM currently has a linear-history policy, which means that merge commits are
-not allowed. The `llvm-project` repo on github is configured to reject pushes
-that include merges, so the `git rebase` step above is required.
+When pushing directly from the command-line to the `main`` branch, you will need
+to rebase your change. LLVM has a linear-history policy, which means
+that merge commits are not allowed and the `main` branch is configured to reject
+pushes that include merges.
+
+GitHub will display a message that looks like:
+
+.. code-block:: console
+
+  remote: Bypassed rule violations for refs/heads/main:
+  remote:
+  remote: - Required status check “buildkite/github-pull-requests” is expected.
+
+This can seem scary, but you likely did nothing wrong: it is intended as a warning
```

Comment:
```suggestion
This can seem scary, but it's not related to your changes: it is intended as a warning
```

Reply:
I'm not sure we can say "it's not related to your changes"? It is related in the sense that the change is triggering the warning and the warning applies to the change. It's just that we would disable this warning if we could!

Reply:
What about:

> This can seem scary, but this is just an artifact of the GitHub setup: it is intended as a warning [...]

Reply:
What I meant is that "the warning itself" is unrelated to the changes, regardless of how it's triggered it doesn't investigate the nature or content of the commits, so it's always unrelated to the changes, only to the push action.

> This can seem scary, but this is just an artifact of the GitHub setup: it is intended as a warning [...]

This is fine too.